File: src/main/java/org/ventanas/Java/Ejercicio11.java
package org.ventanas.Java;
// 11. Fibonacci
// Crear un programa que imprima los primeros N números de la secuencia de Fibonacci (el
// número N lo ingresa el usuario).
import java.util.Scanner;

public class Ejercicio11 {
    public static void execute() {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Dame un número entero: ");
        try {
            int num = Integer.parseInt(scanner.nextLine());
            fibonacci(num);
        } catch (Exception e) {
            System.out.println("Error al insertar el número, debe ser entero.");
        }
    }

    public static void fibonacci(int num) {
        int t1 = 0, t2 = 1;
        for (int i = 1; i <= num; i++) {
            System.out.println(t1);
            int sum = t1 + t2;
            t1 = t2;
            t2 = sum;
        }
    }
}

File: src/main/java/org/ventanas/Kotlin/Ejercicio11.kt
package org.ventanas.Kotlin
// 11. Fibonacci
// Crear un programa que imprima los primeros N números de la secuencia de Fibonacci (el
// número N lo ingresa el usuario).

fun ejercicio11() {
    print("Dame un número entero: ")
    try {
        val num = readln().toInt()

        fibonacci(num)

    }catch (e:Exception){
        println("Error al insertar el numero, debe de ser entero.")
    }
}

fun fibonacci(num : Int) {
    var t1 = 0
    var t2 = 1
    for (i in 1..num) {
        println(t1)
        val sum = t1 + t2
        t1 = t2
        t2 = sum
    }
}
